company-clang.el

```diff
@@ -80,6 +80,76 @@ or automatically through a custom `company-clang-prefix-guesser'."
     (insert-file-contents-literally file nil beg end)
     (buffer-string)))
 
+(defcustom company-clang-parse-documentation nil
+  "When non-nil, clang will parse completion's comments/documentation."
+  :type 'boolean)
+
+(defcustom company-clang-documentation-fill-column 70
+  "Column beyond which automatic line-wrapping should happen."
+  :type 'integer)
+
+(defcustom company-clang-documentation-justify 'full
+  "Specifies which  kind of justification to do."
+  :type '(choice (const :tag "Full" full)
+		 (const :tag "Left" left)
+		 (const :tag "Right" right)
+		 (const :tag "Center" center)
+                 (const :tag "None" none)))
+
+(defvar company-clang-summary-list
+  "Association list of tags and them related documentation.")
+
+(defun company-clang-meta-for-tag (tag)
+  "Extract the meta data of a TAG from `company-clang-summary-list'."
+  (let ((element (assoc tag company-clang-summary-list))
+	(match)
+	(meta))
+    (when element
+      (setq match (car element))
+      (setq meta (get-text-property 0 'meta match))
+      meta)))
+
+(defun company-clang-documentation-for-tag (tag)
+  "Extract the documentation of a TAG from `company-clang-summary-list'."
+  (let ((element (assoc tag company-clang-summary-list))
```

Comment:
Here's the problem: `assoc` will find you a match that's `equal` to TAG. What about function overloads? Different functions can be named the same, as long as they accept different sets of arguments.

These overloads will have the same string value but different `meta` text property, which we add to completions in `company-clang--parse-output`. Adding the documentation strings would probably be too confusing, though. 

(That has just occurred to me) you could the meta values to index `company-clang-summary-list` instead. However, you could also try this approach: for each completion, call `cl-gensym` and put the new symbol on it as the value of `cc-tag` property. Then add `meta` and `doc` to the symbol as its properties, using `put`. Retrieving them would likewise be one `get-text-property` plus one `get`. It's worth testing to see if this indirection's overhead will be noticeable.